scheduler/reconcile.go

```diff
@@ -668,6 +670,14 @@ func (a *allocReconciler) computeStop(group *structs.TaskGroup, nameIndex *alloc
 		return stop
 	}
 
+	// Filter out any terminal allocations from the untainted set
+	// This is so that we don't try to mark them as stopped redundantly
+	for id, alloc := range untainted {
```

Comment:
create a new `filterByTerminalAllocs() (nonTerminal, terminal allocSet)`. You are modifying a shared map in place

Reply:
I saw the rest of this method, like line 693 and 735 also deleting from `untainted` in place. It works because (delete)[https://github.com/golang/go/blob/master/src/runtime/hashmap.go#L511] sets a flag rather than try to rearrange hashmap buckets during iteration. 

Happy to make this change if you think its better for clarity though